app/models/ability.rb

```diff
@@ -206,22 +207,15 @@ def item_abilities
   end
 
   def message_abilities
-    if can_manage_messages?
-      # Supervisors and Reviewers can see all messages on all offers (and items belonging to an offer)
-      # Order fulfillment and Order Administrators are allowed to see all messages on all orders.
-      can %i[index show create], Message
-    else
-      # A donor / charity user is allowed to read any non-private message
-      # that is on an offer / order created by them
-      can [:index, :show], Message, @user.messages do |message|
-        message.related_object&.created_by_id == @user_id && !message.is_private
-      end
-      # A donor / charity user cannot create a private message
-      can :create, Message do |message|
-        message.related_object&.created_by_id == @user_id && !message.is_private
-      end
+    can %i[index show create], Message, messageable_type: 'Offer' if can_manage_offer_messages?
+    can %i[index show create], Message, messageable_type: 'Item' if can_manage_offer_messages?
+    can %i[index show create], Message, messageable_type: 'Order' if can_manage_order_messages?
```

Comment:
@swatijadhav as mentioned in scrum, this might be a good time to add `can_manage_package_messages` permissions in your related PR.
cc @bharat619